### ISSUE TITLE
refactor: inject http.Client into CheckExternalLinks to enable httptest.NewTestServer

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#410]
## Overview

`CheckExternalLinks` constructs its own `*http.Client` internally, making it impossible to inject a test client from outside.
Go 1.27 introduced `httptest.NewTestServer`, whose `Client()` routes requests to the in-process server transparently — so tests can use real production URLs instead of replacing them with the test server address.
To take advantage of this, the function must accept a client from the caller.

## Problem

```go
// internal/rule/external_link.go
client := &http.Client{
    Timeout: time.Duration(timeoutSeconds) * time.Second,
}
```

Because `client` is created inside `CheckExternalLinks`, tests are forced to rewrite every URL in the test markdown to point to the temporary localhost server:

```go
markdown := fmt.Sprintf(`[ok](%s/ok)`, ts.URL)  // real URL is lost
```

This means tests cannot verify that production code actually requests the correct host or path.

## Proposed change

Add a `*http.Client` parameter to `CheckExternalLinks`.
When `nil` is passed, the function falls back to its current behaviour (constructs a default client using `timeoutSeconds`).

```go
func CheckExternalLinks(
    path string, ctx *preprocess.Context, offset int,
    skipPatterns []*regexp.Regexp,
    timeoutSeconds, retryDelayMs, maxConcurrency, maxRetries int,
    allowedStatuses []int, urlCache *sync.Map,
    perHostConcurrency, perHostIntervalMs int,
    client *http.Client,   // nil → build default
) ([]LintError, int)
```

### Effect on tests (Go 1.27+)

```go
ts := httptest.NewTestServer(handler)
// ts.Client() routes any host to the in-process server
results, _ := rule.CheckExternalLinks(
    "mock.md", ctx, 0,
    nil, 10, 10, rule.DefaultMaxConcurrency, rule.DefaultMaxRetries,
    nil, &sync.Map{}, 0, 0,
    ts.Client(),   // injected
)
```

Markdown in tests can now contain literal production-like URLs (`https://example.com/ok`) while still being served by the test server — letting tests assert on method, host, and path correctness.

### Effect on the linter (production)

```go
// internal/linter/linter.go — pass nil; behaviour unchanged
rule.CheckExternalLinks(..., nil)
```

## References

- Go 1.27 release notes: `net/http/httptest.NewTestServer`
- https://zenn.dev/catatsuy/articles/8e8dbf6afd1196

**Comment by shinagawa-web** [comment]
## Implementation Plan

### Files to modify

- `internal/rule/external_link.go` — add `client *http.Client` parameter; build default client when `nil`
- `internal/linter/linter.go` — pass `nil` at the call site (`linter.go:412`)
- `internal/rule/external_link_test.go` — pass `nil` for now; switch to `ts.Client()` after Go 1.27 bump
- `internal/rule/external_link_network_test.go` — same as above

### Checklist

- [ ] Update `CheckExternalLinks` signature and nil-guard
- [ ] Update `linter.go` call site
- [ ] Update all test call sites to pass `nil` (no behaviour change)
- [ ] Bump `go.mod` to Go 1.27
- [ ] Rewrite tests to use `httptest.NewTestServer` + `ts.Client()`
- [ ] `make test-all`
